### DISCUSSION TITLE
Securing client connection failed

### DISCUSSION THREAD

**Discussion by crazyn2** [dstotijn/hetty#83] [Q&A] [answered]
Hey,everyone.I got some errors when running hetty in docker container.
Operating System:Ubuntu20.04
Docker: 20.10.7
Image: dstotijn/hetty:latest
It's my steps to configure hetty.

1.docker run -v $HOME/.hetty:/root/.hetty -p 8080:8080 --name hetty dstotijn/hetty
2.sudo cp ~/.hetty/hetty_cert.pem /usr/local/share/ca-certificates/hetty.crt && sudo update-ca-certificates
2.open http://localhost:8080/ and create a new project
3.add Foxyproxy http proxy in Firefox browser
4.open a https website.

Error:
2021/07/16 05:05:45 [ERROR] Securing client connection failed: handshake error: remote error: tls: bad certificate

By the way, I made a trival to generate certificate by openssl mentioned in official manual but it didn't work.

Could anyone fix the error?

**Comment by dstotijn** [comment, marked as answer]
Hard to tell the root cause from the error, especially if the cert was manually created via openssl.
What happens if you remove `~/.hetty/hetty_cert.pem` and ~`/.hetty/hetty_key.pem`, then run the 4 steps again?
When running `hetty`, it should then autogenerate a cert keypair.

**Reply by crazyn2** [reply]
I‘ve avoided this problem by importing pem into browser manually.While reading the document of mitmproxy,I noticed that pem file should be converted into crt type via openssl  not just modify the subfix between crt and pem

**Reply by dstotijn** [reply]
Importing `hetty_cert.pem` in Firefox should work (tested on Firefox v98 on macOS 12.1). Note that Firefox uses its own trust store, thus installing to the system’s root store has no effect. See https://github.com/dstotijn/hetty/issues/101#issuecomment-1067215192.
